File: write-api-server-kotlin/src/main/kotlin/com/github/j5ik2o/adceet/api/write/use/case/AddMessageUseCase.kt
package com.github.j5ik2o.adceet.api.write.use.case

import com.github.j5ik2o.adceet.api.write.domain.Message
import com.github.j5ik2o.adceet.api.write.domain.ThreadId
import java.util.concurrent.CompletionStage

interface AddMessageUseCase {
  fun execute(message: Message): CompletionStage<ThreadId>
}

File: write-api-server-scala/src/main/scala/com/github/j5ik2o/adceet/api/write/use/case/AddMessageUseCase.scala
package com.github.j5ik2o.adceet.api.write.use.`case`

import com.github.j5ik2o.adceet.api.write.domain.{ Message, ThreadId }

import scala.concurrent.{ ExecutionContext, Future }

trait AddMessageUseCase {

  def execute(message: Message)(implicit ec: ExecutionContext): Future[ThreadId]

}
